packages/core/src/nodeParsers/utils.ts

```diff
@@ -0,0 +1,75 @@
+import _ from "lodash";
+import {
+  BaseNode,
+  Document,
+  ImageDocument,
+  NodeRelationship,
+  TextNode,
+} from "../Node";
+
+type TextSplitter = (s: string) => string[];
+
+/**
+ * Splits the text of a document into smaller parts.
+ * @param document - The document to split.
+ * @param textSplitter - The text splitter to use.
+ * @returns An array of text splits.
+ */
+function getTextSplitsFromDocument(
+  document: Document,
+  textSplitter: TextSplitter,
+) {
+  const text = document.getText();
+  return textSplitter(text);
+}
+
+/**
+ * Generates an array of nodes from a document.
+ * @param doc
+ * @param textSplitter - The text splitter to use.
+ * @param includeMetadata - Whether to include metadata in the nodes.
+ * @param includePrevNextRel - Whether to include previous and next relationships in the nodes.
+ * @returns An array of nodes.
+ */
+export function getNodesFromDocument(
+  doc: BaseNode,
+  textSplitter: TextSplitter,
+  includeMetadata: boolean = true,
+  includePrevNextRel: boolean = true,
+): TextNode[] {
+  if (doc instanceof ImageDocument) {
+    // TODO: use text splitter on text of image documents
+    return [doc];
+  }
+  if (!(doc instanceof Document)) {
+    throw new Error("Expected either an Image Document or Document");
+  }
+  const document = doc as Document;
+  const nodes: TextNode[] = [];
+
+  const textSplits = getTextSplitsFromDocument(document, textSplitter);
+
+  textSplits.forEach((textSplit) => {
+    const node = new TextNode({
+      text: textSplit,
+      metadata: includeMetadata ? _.cloneDeep(document.metadata) : {},
```

Comment:
@yisding fun fact: we also had here just `document.metadata` before, which means that the existing NodeParser is working on the same metadata